### ISSUE TITLE
bug(matTooltip): tooltip doesn't work with ChangeDetectionStrategy.OnPush

### ISSUE THREAD

**Issue by PashaTurok** [angular/components#24959]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

I have a table with tooltips inside rows:

```
 <ng-container matColumnDef="icons">
    <th mat-header-cell *matHeaderCellDef>#</th>
    <td mat-cell *matCellDef="let element">
      <span [matTooltip]="'Hello, world'" matTooltipPosition="below"
        >tooltip</span>
    </td>
  </ng-container>
```

The table can have thousands of rows, that's why I have to use ChangeDetectionStrategy.OnPush. However, with such strategy tooltips are shown without text.

### Reproduction

StackBlitz demo is [here](https://stackblitz.com/edit/tooltip-and-changedetectionstrategy-onpush?file=src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.component.ts)

### Expected Behavior

Tooltips must be shown with text.

### Actual Behavior

Tooltips are shown without text

### Environment

- Angular: 13.3.9
- CDK/Material: 13.3.7
- Browser(s): FF 95
- Operating System (e.g. Windows, macOS, Ubuntu): Ubuntu

**Comment by crisbeto** [comment]
It doesn't work, because you're detaching the `ChangeDetectorRef` in the constructor which is going to disable change detection for the entire app.

**Comment by PashaTurok** [comment]
@crisbeto  I detach it because I use `onPush` strategy. If I don't call `this._changeDetectorRef.detach();` then angular doesn't work in `onPush` mode.

**Comment by crisbeto** [comment]
`OnPush` change detection is activated when you set `changeDetection: ChangeDetectionStrategy.OnPush`, but that doesn't mean that change detection will be disabled completely, otherwise nothing would ever happen in the app.

**Comment by PashaTurok** [comment]
@crisbeto Thank you very much for your explanation. This is a modified version of the [demo](https://stackblitz.com/edit/tooltip-and-changedetectionstrategy-onpush-npkmqn?file=src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.component.ts) with three modifications:

- `detach()` is commented
- added `fooTest` getter in .ts `console.log('change was checked');  return true;`
- added `{{fooTest}} in .html

With these modifications when you move mouse over tooltip you see in console `change was checked`. So, when we don't call `datach()` method in constructor angular continues to check view/model. And when I have thousands of rows and any event is processed (keypress, mouse wheel etc) I see serious performance problems although nor view nor model has been change.

**Comment by crisbeto** [comment]
That happens because the tooltip is listening to some events which will trigger change detection. Change detection in itself isn't slow (it's basically a bunch of `===` expressions), it's what your app does in response to it is what causes the slowdown. I'm closing the issue because the tooltip is working as expected.
